client/src/nv_ingest_client/message_clients/rest/rest_client.py

```diff
@@ -157,38 +160,42 @@ def fetch_message(self, job_id: str, timeout: float = 10) -> Optional[str]:
 
         Returns
         -------
-        Optional[str]
-            The fetched message, or None if no message could be fetched.
-
-        Raises
-        ------
-        ValueError
-            If fetching the message fails after the specified number of retries or due to other critical errors.
+        ResponseSchema
+            The fetched message wrapped in a ResponseSchema object.
         """
         retries = 0
         while True:
             try:
                 # Fetch via HTTP
                 url = f"{self.generate_url(self._host, self._port)}{self._fetch_endpoint}/{job_id}"
                 logger.debug(f"Invoking fetch_message http endpoint @ '{url}'")
-                result = requests.get(url)
+                result = requests.get(url, timeout=self._connection_timeout)
 
                 response_code = result.status_code
                 if response_code in _TERMINAL_RESPONSE_STATUSES:
-                    # Any terminal response code results in a RuntimeError
-                    raise RuntimeError(f"A terminal response code: {response_code} was received \
-                                       when fetching JobSpec: {job_id} with server response \
-                                        '{result.text}'")
+                    # Terminal response code; return error ResponseSchema
+                    return ResponseSchema(
+                        response_code=1,
+                        response_reason=f"Terminal response code {response_code} received when fetching JobSpec: {job_id}",
+                        response=result.text
+                    )
                 else:
                     # If the result contains a 200 then return the raw JSON string response
                     if response_code == 200:
-                        return result.text
+                        return ResponseSchema(
+                            response_code=0,
+                            response_reason="OK",
+                            response=result.text,
```

Comment:
The ResponseSchema has entries for the trace_id, and the transaction_id (see below). I think the reason they don't show up here is because this is the 'fetch' side, which doesn't do any tracing.  The `submit_message` path below this does add the trace/trasaction ids.

```python
class ResponseSchema(BaseModel):
    response_code: int
    response_reason: Optional[str] = "OK"
    response: Union[str, dict, None] = None
    trace_id: Optional[str] = None  # Unique trace ID
    transaction_id: Optional[str] = None  # Unique transaction ID
```